Feat(web): FloatingUI integration to Tooltip #DS-1086 (#1197)
lmc-eu/spirit-design-system: packages/web/src/scss/components/Tooltip/README.md

</div>
```

> ### Floating UI functionality

Reviewer: ```suggestion
### Advanced floating functionality
```

I think nobody cares for underlying implementation.

Author: changed, thank you

> ### Floating UI functionality

Author: fixed

> To enable the Floating UI functionality, you need to activate the JS plugin, wrap your tooltip with an element having the `data-spirit-element="tooltip-wrapper"` data attribute, and add the `data-spirit-placement-controlled` attribute to your tooltip element.

Reviewer: ```suggestion
To enable the Floating UI functionality, you need to activate the JS plugin, wrap your tooltip with an element having the `data-spirit-element="tooltip-wrapper"` data attribute, and add the `data-spirit-placement-controlled` attribute to your tooltip element to modify the styling of arrows.
```

Author: changed, thank you

> To enable the Floating UI functionality, you need to activate the JS plugin, wrap your tooltip with an element having the `data-spirit-element="tooltip-wrapper"` data attribute, and add the `data-spirit-placement-controlled` attribute to your tooltip element.

Reviewer 2: Wasn't it four steps? If yes, I'd put the instructions as a numbered list (three steps are OK in a sentence).

Author: Added 4th step - info about feature flag

```html
<div class="spirit-feature-tooltip-enable-data-placement">
  <div class="TooltipWrapper d-inline-block" data-spirit-element="tooltip-wrapper">
    <button type="button" id="tooltip-trigger" data-spirit-toggle="tooltip" data-spirit-target="#floating-ui-example">
      Toggle tooltip
    </button>
    <div
      id="floating-ui-example"
      class="Tooltip is-hidden"
      data-spirit-placement="bottom"
      data-spirit-flip="true"
      data-spirit-flip-cross-axis="true"
      data-spirit-flip-fallback-placements="bottom, right, top"
      data-spirit-placement-controlled
    >
      Hello there!
      <span class="Tooltip__arrow" data-spirit-element="arrow"></span>
    </div>
  </div>
</div>

> ```

Reviewer 2: This is cool, I'd just add a very basic example before this one, showing just the things that are _essential_ to launch a Floating-UI tooltip.

Author: I left only a basic example there because the example with more options is almost the same, just with additional data attributes listed in the table below.

## Floating UI Attributes

| Attribute                                       | Type                                         | Default  | Required | Description                                                                                                                                                                                                                                                             |
| ----------------------------------------------- | -------------------------------------------- | -------- | -------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `data-spirit-flip-cross-axis`                   | ["true" \| "false"]                          | "false"  | ✕        | This is the [cross axis][floating-ui-flip-crossaxis] in which overflow is checked to perform a flip, the axis perpendicular to main axis                                                                                                                                |

> | `data-spirit-flip-fallback-axis-sids-direction` | ["none" \| "start" \| "end" ]                | "none"   | ✕        | Whether to allow [fallback to the opposite axis][floating-ui-flip-fallbackaxissidedirection] if no placements along the preferred placement axis fit, and if so, which side direction along that axis to choose. If necessary, it will fallback to the other direction. |

Reviewer 2: ```suggestion
| `data-spirit-flip-fallback-axis-side-direction` | ["none" \| "start" \| "end" ]                | "none"   | ✕        | Whether to allow [fallback to the opposite axis][floating-ui-flip-fallback-axis-side-direction] if no placements along the preferred placement axis fit, and if so, which side direction along that axis to choose. If necessary, it will fallback to the other direction. |
```

?

Author: fixed

> | `data-spirit-flip-fallback-placements`          | string                                       | -        | ✕        | This describes an [explicit placements][floating-ui-flip-fallbackplacements] to try if the initial placement doesn’t fit on the axes in which overflow is checked. For example you can set `"top, right, bottom"`                                                       |

Reviewer 2: ```suggestion
| `data-spirit-flip-fallback-placements`          | string                                       | -        | ✕        | This describes a list of [explicit placements][floating-ui-flip-fallbackplacements] to try if the initial placement doesn’t fit on the axes in which overflow is checked. For example you can set `"top, right, bottom"`                                                       |
```

? Because there is `an` with plural…

Author: fixed

| `data-spirit-placement`                         | [Placement Dictionary][dictionary-placement] | "bottom" | ✕        | Placement of tooltip                                                                                                                                                                                                                                                    |
| `data-spirit-shift`                             | ["true" \| "false"]                          | "false"  | ✕        | Turn on/off [shift functionality][floating-ui-shift]                                                                                                                                                                                                                    |
| `data-spirit-size`                              | ["true" \| "false"]                          | "false"  | ✕        | Turn on/off [size functionality][floating-ui-size]                                                                                                                                                                                                                      |

> 👆 All the attributes mentioned above can be set as an object in the `config` attribute, like this: `data-spirit-config='{"flip": "true", "flipFallbackPlacements": "top, right, bottom"}'`. Please note that this configuration has lower priority than individual attributes and will be overwritten by them.

Reviewer 2: Are the quotes around `true` correct?

Author: quotes removed

tooltip.hide();
```

[dictionary-placement]: https://github.com/lmc-eu/spirit-design-system/tree/main/docs/DICTIONARIES.md#placement
[example]: https://spirit-design-system-demo.netlify.app/src/scss/components/tooltip/#advanced-positioning

> [floating-ui-flip-crossaxis]: https://floating-ui.com/docs/flip#crossaxis
> [floating-ui-flip-fallbackaxissidedirection]: https://floating-ui.com/docs/flip#fallbackaxissidedirection
> [floating-ui-flip-fallbackplacements]: https://floating-ui.com/docs/flip#fallbackplacements

Reviewer 2: I'd rather stick to hyphenation even in the parameter names, it's much more readable:

```suggestion
[floating-ui-flip-cross-axis]: https://floating-ui.com/docs/flip#crossaxis
[floating-ui-flip-fallback-axis-side-direction]: https://floating-ui.com/docs/flip#fallbackaxissidedirection
[floating-ui-flip-fallback-placements]: https://floating-ui.com/docs/flip#fallbackplacements
```

Author: hyphenation added for better readabilty

[floating-ui-flip]: https://floating-ui.com/docs/flip
[floating-ui-shift]: https://floating-ui.com/docs/shift
[floating-ui-size]: https://floating-ui.com/docs/size
[floating-ui]: https://floating-ui.com
[readme-deprecations]: https://github.com/lmc-eu/spirit-design-system/tree/main/packages/web/README.md#deprecations

> [readme-feature-flags]: https://github.com/lmc-eu/spirit-design-system/tree/main/packages/web/README.md#feature-flags

Reviewer 2: These two are going to be `blob` 🙂.

Author: changed to blob
